aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+import com.amplifyframework.auth.result.step.AuthNextSignInStep;
+import com.amplifyframework.auth.result.step.AuthNextSignUpStep;
+import com.amplifyframework.auth.result.step.AuthResetPasswordStep;
+import com.amplifyframework.auth.result.step.AuthSignInStep;
+import com.amplifyframework.auth.result.step.AuthSignUpStep;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.client.Callback;
+import com.amazonaws.mobile.client.UserState;
+import com.amazonaws.mobile.client.UserStateDetails;
+import com.amazonaws.mobile.client.results.ForgotPasswordResult;
+import com.amazonaws.mobile.client.results.ForgotPasswordState;
+import com.amazonaws.mobile.client.results.SignInResult;
+import com.amazonaws.mobile.client.results.SignInState;
+import com.amazonaws.mobile.client.results.SignUpResult;
+import com.amazonaws.mobile.client.results.UserCodeDeliveryDetails;
+import com.amazonaws.mobile.config.AWSConfiguration;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.Map;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test that the current implementation of Auth as a wrapper of AWSMobileClient calls the correct
+ * AWSMobileClient methods with the correct parameters when the different Auth methods are called.
+ */
+@RunWith(RobolectricTestRunner.class)
+@SuppressWarnings("unchecked")
+public final class AuthComponentTest {
+    private AWSMobileClient mobileClient;
+    private AWSCognitoAuthPlugin authPlugin;
+    private final String userId = "myId";
+    private final String destination = "e***@email.com";
+    private final String deliveryMedium = "sms";
+    private final String attributeName = "email";
+    private final String username = "username";
+    private final String password = "password123";
+    private final String newPassword = "newPassword123";
+    private final String attributeKey = AuthUserAttributeKey.email().getKeyString();
+    private final String attributeVal = "email@email.com";
+    private final String confirmationCode = "confirm";
+    private final Map<String, String> metadata = Collections.singletonMap("aCustomKey", "aCustomVal");
+
+    @Before
+    public void setup() throws AmplifyException {
+        mobileClient = mock(AWSMobileClient.class);
+        authPlugin = new AWSCognitoAuthPlugin(mobileClient, userId);
```

Comment:
Were you able to do:
```
AuthCategory auth = new AuthCategory();
auth.addPlugin(new AWSCognitoAuthPlugin());
CategoryConfiguration config = // blah blach
auth.configure(config, getApplicationContext());
auth.initialize(getApplicationContext());
```
And then test against the category? You'd get more coverage that way.

Testing a thin veneer of invocation verifications on just the plugin is going to be a little brittle.

Reply:
Oh yeah, I was going to update later to try that - let me see.